src/Element/FEAElement_Triangle3_membrane.cpp

```diff
@@ -40,51 +40,42 @@ void FEAElement_Triangle3_membrane::buildBasis( const IQuadPts * const &quad,
     R[qua*3 + 2] = qua_s;
   }
 
-  const double dx_dr = ctrl_x[0] * (-1.0) + ctrl_x[1];
-  const double dy_dr = ctrl_y[0] * (-1.0) + ctrl_y[1];
-  const double dz_dr = ctrl_z[0] * (-1.0) + ctrl_z[1];
-  const double dx_ds = ctrl_x[0] * (-1.0) + ctrl_x[2];
-  const double dy_ds = ctrl_y[0] * (-1.0) + ctrl_y[2];
-  const double dz_ds = ctrl_z[0] * (-1.0) + ctrl_z[2];
+  const Vector_3 dx_dr(ctrl_x[0] * (-1.0) + ctrl_x[1],
+    ctrl_y[0] * (-1.0) + ctrl_y[1],
+    ctrl_z[0] * (-1.0) + ctrl_z[1]);
 
-  // vec(un) = vec(dx_dr) x vec(dx_ds)
-  MATH_T::cross3d(dx_dr, dy_dr, dz_dr, dx_ds, dy_ds, dz_ds, unx, uny, unz);
+  const Vector_3 dx_ds(ctrl_x[0] * (-1.0) + ctrl_x[2],
+    ctrl_y[0] * (-1.0) + ctrl_y[2],
+    ctrl_z[0] * (-1.0) + ctrl_z[2]);
 
-  MATH_T::normalize3d( unx, uny, unz );
+  // vec(un) = vec(dx_dr) x vec(dx_ds)
+  un = cross_product( dx_dr, dx_ds );
+  un.normalize();
```

Comment:
it may need also a normalize( Vector_3 &vec) function, then un = normalize( cross_product(dx_dr, dx_ds) ), which I have checked.

Reply:
Let’s keep it in the current way.